src/main/java/core/basesyntax/RightTriangle.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+public class RightTriangle extends Figure {
+    private final double firstLeg;
+    private final double secondLeg;
+    private final ColorSupplier color;
+
+    public RightTriangle(ColorSupplier color, double firstLeg, double secondLeg) {
```

Comment:
Creating a figure, don't pass expressions in the constructor. Assign the color inside the constructor using `color.getRandomColor()` and store it as a `Color` type field in the `Figure` class.